lib/contentTypeParser.js

```diff
@@ -25,11 +25,19 @@ const {
 const warning = require('./warnings')
 
 function ContentTypeParser (bodyLimit, onProtoPoisoning, onConstructorPoisoning) {
-  this[kDefaultJsonParse] = getDefaultJsonParser(onProtoPoisoning, onConstructorPoisoning)
+  this.nativeParsers = {
+    'application/json': getDefaultJsonParser(onProtoPoisoning, onConstructorPoisoning),
+    'text/plain': defaultPlainTextParser
+  }
```

Comment:
Those are two elements. I would try to avoid a for loop for two elements and this wrapping object for two elements.

Reply:
Well, it took me quite some time to figure out how the module worked so I thought to reduce the mental puzzle by abstracting it out.

If you ever feel the need to add a third native parser then this would work right out of the box.I   I assumed (maybe incorrectly) that the JIT will do its magic here anyway.

But if you really don't want the object and the loop then I can pull it out again. As the easy fix is just to rename the original hasParser to existingParser and add the new hasParser without the `if` statements.
(that will need to wait a week before I'm behind my dev machine again)

Reply:
ok!

Reply:
FYI: just out of curiosity: I decided to check performance differences and purely on the loop vs separate statements its about 10% difference. See https://jsben.ch/tgJHe